pkg/stanza/operator/transformer/assign_keys/assign_keys.go

```diff
@@ -0,0 +1,132 @@
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/entry"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/errors"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/stanza/operator/helper"
+	"go.uber.org/zap"
+)
+
+const operatorType = "assign_keys"
+
+func init() {
+	operator.Register(operatorType, func() operator.Builder { return NewConfig() })
+}
+
+// NewConfig creates a new assign_keys operator config with default values
+func NewConfig() *Config {
+	return NewConfigWithID(operatorType)
+}
+
+// NewConfigWithID creates a new assign_keys operator config with default values
+func NewConfigWithID(operatorID string) *Config {
+	return &Config{
+		TransformerConfig: helper.NewTransformerConfig(operatorID, operatorType),
+	}
+}
+
+// Config is the configuration of a assign_keys operator
+type Config struct {
+	helper.TransformerConfig `mapstructure:",squash"`
+	Field                    entry.Field `mapstructure:"field"`
+	Keys                     []string    `mapstructure:"keys"`
+}
+
+// Build will build an assign_keys operator from the supplied configuration
+func (c Config) Build(logger *zap.SugaredLogger) (operator.Operator, error) {
+	transformerOperator, err := c.TransformerConfig.Build(logger)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(c.Keys) == 0 {
+		return nil, fmt.Errorf("assign_keys missing required field keys")
+	}
+
+	if e, ok := c.Field.FieldInterface.(entry.BodyField); ok {
+		return &Transformer[entry.BodyField]{
+			TransformerOperator: transformerOperator,
+			Field:               e,
+			Keys:                c.Keys,
+		}, nil
+	}
+
+	if e, ok := c.Field.FieldInterface.(entry.ResourceField); ok {
+		return &Transformer[entry.ResourceField]{
+			TransformerOperator: transformerOperator,
+			Field:               e,
+			Keys:                c.Keys,
+		}, nil
+	}
+
+	if e, ok := c.Field.FieldInterface.(entry.AttributeField); ok {
+		return &Transformer[entry.AttributeField]{
+			TransformerOperator: transformerOperator,
+			Field:               e,
+			Keys:                c.Keys,
+		}, nil
+	}
+
+	return nil, fmt.Errorf("invalid field type: %T", c.Field.FieldInterface)
+}
+
+// Transformer transforms a list in the entry field into a map. Each value is assigned a key from configuration keys
+type Transformer[T interface {
+	entry.BodyField | entry.ResourceField | entry.AttributeField
+	entry.FieldInterface
+}] struct {
+	helper.TransformerOperator
+	Field T
+	Keys  []string
+}
+
+// Process will process an entry with AssignKeys transformation.
+func (p *Transformer[T]) Process(ctx context.Context, entry *entry.Entry) error {
+	return p.ProcessWith(ctx, entry, p.Transform)
+}
+
+// Transform will apply AssignKeys to an entry
+func (p *Transformer[T]) Transform(entry *entry.Entry) error {
+	inputListInterface, ok := entry.Delete(p.Field)
```

Comment:
We follow a pattern where parsing and transformation operations should not be destructive, by default. It is a little more verbose for the user, but the original field should be left in place. If the user chooses to `parse_to` the same field as `parse_from`, it will naturally overwrite it. Otherwise, they can delete the field explicitly with the `remove` operator.

This means we should use `Get` here and shouldn't need any rollback behavior.

Reply:
Done